baybe/surrogates/gaussian_process/prior_modules.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+from typing import Any
+from torch import Tensor
+
+from copy import deepcopy
+
+import gpytorch
+import torch
+from botorch.models import SingleTaskGP
+
+
+class PriorMean(gpytorch.means.Mean):
+    """GPyTorch mean module using a trained GP as prior mean.
+
+    This mean module wraps a trained Gaussian Process and uses its predictions
+    as the mean function for another GP.
+    """
+
+    def __init__(
+        self, gp: SingleTaskGP, batch_shape: torch.Size = torch.Size(), **kwargs: Any
+    ) -> None:
+        """Initialize the GP-based mean module.
+
+        Args:
+            gp: Trained Gaussian Process to use as mean function.
+            batch_shape: Batch shape for the mean module.
+            **kwargs: Additional keyword arguments.
+        """
+        super().__init__()
+        # See https://github.com/cornellius-gp/gpytorch/issues/743
+        self.gp: SingleTaskGP = deepcopy(gp)
+        self.batch_shape: torch.Size = batch_shape
+        for param in self.gp.parameters():
+            param.requires_grad = False
+
+    def reset_gp(self) -> None:
```

Comment:
When I was developing this using this (besides other code bits) that "freeze" the parameters and use eval mode was needed to make sure model is used correctly. However, I am not 100% sure this is the best way to do it. If there is time maybe it could be investigated if these resets could be made nicer (e.g. do them only during init).